include/deal.II/base/work_stream.h

```diff
@@ -663,6 +663,163 @@ namespace WorkStream
+     * scheduling rather than TBB. Currently does not support chunking.
+     */
+
+    namespace taskflow_no_coloring
+    {
+      template <typename Worker,
+                typename Copier,
+                typename Iterator,
+                typename ScratchData,
+                typename CopyData>
+
+      /**
+       *  The last two arguments in this function are for chunking support which
+       * currently does not exist but ideally will later. For now they are
+       * ignored but still here to permit existing programs to function
+       */
+      void
+      run(const Iterator                          &begin,
+          const typename identity<Iterator>::type &end,
+          Worker                                   worker,
+          Copier                                   copier,
+          const ScratchData                       &sample_scratch_data,
+          const CopyData                          &sample_copy_data,
+          const unsigned int queue_length = 2 * MultithreadInfo::n_threads(),
+          const unsigned int chunk_size   = 8)
+
+      {
+        struct ScratchDataObjects
+        {
+          std::unique_ptr<ScratchData> scratch_data;
+          bool                         currently_in_use;
+
+          /**
+           * Default constructor.
+           */
+          ScratchDataObjects()
+            : currently_in_use(false)
+          {}
+
+          ScratchDataObjects(std::unique_ptr<ScratchData> &&p,
+                             const bool                     in_use)
+            : scratch_data(std::move(p))
+            , currently_in_use(in_use)
+          {}
+
+          // Provide a copy constructor that actually doesn't copy the
+          // internal state. This makes handling ScratchAndCopyDataObjects
+          // easier to handle with STL containers.
+          ScratchDataObjects(const ScratchDataObjects &)
+            : currently_in_use(false)
+          {}
+        };
+
+        tf::Executor &executor = MultithreadInfo::get_taskflow_executor();
+        tf::Taskflow  taskflow;
+
+        using ScratchDataList = std::list<ScratchDataObjects>;
+
+        Threads::ThreadLocalStorage<ScratchDataList> data;
+
+        tf::Task last_copier;
+
+        // Silence unused variable arguments.
+        (void)queue_length;
+        (void)chunk_size;
+
+        // This is used to connect each worker to its copier as communication
+        // between tasks is not supported.
+        unsigned int idx = 0;
+
+        std::vector<std::unique_ptr<CopyData>> copy_datas;
+
+        for (Iterator i = begin; i != end; ++i, ++idx)
+          {
+            copy_datas.emplace_back();
+            // Create a worker task.
+            auto worker_task =
+              taskflow
+                .emplace([it = i,
+                          idx,
+                          &data,
+                          &sample_scratch_data,
+                          &sample_copy_data,
+                          &copy_datas,
```

Comment:
This is not going to be thread-safe. You are taking a reference to `copy_datas`, which you access via `copy_datas[idx]` below from the worker thread, but at the same time the thread that spawns all of these tasks is modifying `copy_datas` in line 745. 

(Or perhaps what you are saying is that you are only *creating* the task objects here, not yet running them, and all of the accesses once the tasks are running only ever modify a single entry of the vector. That is ok, but it would be nice to describe this in a comment.)